feat (association-select): hiding and enable/disable of placeholder option, as well as custom text

go4cas opened the pull request:
Provide the developer with options to show/hide and enable/disable the placeholder option
, as well as overriding the default display text value.

Backwards compatible, and fixing issues #169 and #170

go4cas committed:
feat (association-select): allow hiding and enable/disable of placeholder (value=0) option, as well as custom display text

Provide the developer with options to show/hide and enable/disable the placeholder option
, as well as overriding the default display text value.

Backwards compatible, and fixing issues #169 and #170

doktordirk commented on src/component/association-select.js:
```diff
@@ -28,6 +28,12 @@ export class AssociationSelect {
 
+  @bindable hidePlaceholder = false;
+
+  @bindable allowPlaceholderSelect = false;
```
how about 'selectablePlaceholder' (my preference) or 'isPlaceholderSelectable'

doktordirk commented on src/component/association-select.html:
```diff
@@ -1,6 +1,6 @@
 <template>
   <select class="form-control" value.bind="value" multiple.bind="multiple">
-    <option selected disabled value="0">- Select a value -</option>
+    <option selected disabled.bind="!hidePlaceholder && !allowPlaceholderSelect" value="0" show.bind="!hidePlaceholder">${placeholderText || '- Select a value -'}</option>
```
has 'disabled.bind="!hidePlaceholder'  any effect?  hmm guess so.

doktordirk commented:
well, concise and user-friendly. i like it

doktordirk commented:
you accidentally commited the package.json now

go4cas commented:
Uhm, how do I undo that?

doktordirk commented:
depends on your tool of choice. but you can just commit the previous version on top of this

doktordirk commented:
hmm, that didn't work well. now it's back at allowPlaceholderSelect 
try:
git reset --hard HEAD~2

change again allowPlaceholderSelect to selectablePlaceholder

git add -A
git commit --amend --no-edit
git push origin HEAD -f

and then if you're nice you'll update /doc/component.md :smile:

doktordirk commented on doc/components.md:
```diff
@@ -25,6 +25,43 @@ Get all entries of the view-models 'userRepository' repository property and popu
+
+The same as the basic example above, with:
+
+* hiding the placeholder option (`value=0`):
```
value=0 associates someone there is a value=0 to set. maybe better value===0

doktordirk replied:
though thinking of it. maybe just mention once or twice at the right place that placeholder means value 0 and then just use placeholder. that'll make it more readable

doktordirk commented:
ah that worked
now you can just add changes
so eg when chaining the readme commit with
doc(README): xyz

no need to use the old message again, since that would already be logged in the changelog

go4cas committed:
doc(association-select): adding examples for placeholder options

Adding examples for all attributes (hidePlaceholder, selectablePlaceholder, placeholderText) used for setting the placeholder options.

Backwards compatible, and fixing issues #169 and #170

doktordirk commented:
alright. tx: nice job!! let's get that in then
